src/analysis/processing/qgsalgorithmcheckgeometryarea.cpp

```diff
@@ -63,7 +63,7 @@ QString QgsGeometryCheckAreaAlgorithm::shortHelpString() const
 
 Qgis::ProcessingAlgorithmFlags QgsGeometryCheckAreaAlgorithm::flags() const
 {
-  return QgsProcessingAlgorithm::flags() | Qgis::ProcessingAlgorithmFlag::NoThreading;
+  return QgsProcessingAlgorithm::flags() | Qgis::ProcessingAlgorithmFlag::RequiresProject;
```

Comment:
I don't think these algorithms ARE thread safe -- at least, there's access to the project from the process step, which hints that there's at least some logic which would need to be moved to the prepare stage. Can we leave the no thread flag for now and then address that in a follow up?